celery_utils/logged_task.py

```diff
@@ -0,0 +1,34 @@
+"""
+Improved logging for celery tasks.
+"""
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+
+from celery import Task
+
+
+log = logging.getLogger(__name__)
+
+
+# pylint: disable=abstract-method
+class LoggedTask(Task):
+    """
+    Task base class that emits a log statement when it gets submitted.
+    """
+
+    abstract = True
+
+    def apply_async(self, args, kwargs, **options):
+        """
+        Emit a log statement when the task is submitted.
+        """
+        result = super(LoggedTask, self).apply_async(args, kwargs, **options)
+        log.info('Task {}[{}] submitted with arguments {}, {}'.format(
+            self.name,
+            result.id,
+            args,
+            kwargs
+        ))
```

Comment:
So this captures the time when the task is created.

Two questions:
1. Will it be easy to find how long a task has been waiting in the queue if you don't track the time when the task was actually run?  We can possibly do this for grades - but it would involve checking for 2 different types of log statements.

2. In the ticket, we wrote that we also wanted to know the stack trace of the root exception when a retry happens.  Did you verify that this is already captured?  (I thought it wasn't.). If not, can we capture that in this base class?